contributed_definitions/nyaml/SPM/NXspm.yaml

```diff
@@ -0,0 +1,832 @@
+        (NXscan_control):
+          doc: |
+            The scan control information like scan region or phase space, type of scan (e.g.
+            mesh, spiral, etc.), and scan speed, etc. This group mainly stores the raw scan
+            data. For processed data or final experimental data would go to NXdata group.
+          scan_name:
+            exists: optional
+            doc: |
+              The name of the scan, mainly lab defined, that helps to differentiate among different
+              scans performed in lab. This field is intended for several types of scan control
+              run under the same environment.
+      current_sensor(NXsensor):
+        exists: optional
+        doc: |
+          Information for current sensor.
+        (NXamplifier):
+          exists: optional
+          doc: |
+            An amplifier information for the input signal (e.g. from tip).
+      voltage_sensor(NXsensor):
+        exists: optional
+        doc: |
+          The sensor information for the voltage device.
+        (NXamplifier):
+          exists: optional
+      piazo_sensor(NXsensor):
+        doc: |
+          The piezo sensor refers to the height (or Z) piezo sensor if nothing is
+          mentioned along the X and Y directions.
+        x(NX_NUMBER):
+          unit: NX_LENGTH
+          doc: |
+            The x position of the piezo. In STS experiment, the piezo stays fixed at
+            x,y and z and the the tunneling current is measured with respect to the
+            bias voltage (sweep voltage).
+        y(NX_NUMBER):
+          unit: NX_LENGTH
+          doc: |
+            The y position of the piezo.
+        z(NX_NUMBER):
+          unit: NX_LENGTH
+          doc: |
+            The z position of the piezo.
+        piezo_configuration(NXpiezo_config_spm):
+          exists: optional
+          doc: |
+            The piezo configuration information like piezoelectric calibration and material
+            properties.
+          calibration(NXcalibration):
+            exists: optional
+            doc: |
+              Calibration of the piezo device.
+        (NXpositioner_spm):
+          doc: |
+            The positioner information like the position of the tip, the position of the
+            sample, PID loop feedback etc.
+          z_controller(NXpid):
+            doc: |
+              The PID controller information for the z-axis.
+            z(NX_NUMBER):
+              doc: |
+                To indicate the relative tip position z between tip and sample. The tip position
+                can also be varied when the z_controller is not running.
+            controller_status(NX_BOOLEAN):
+              doc: |
+                Status if controller is active.
+        piezo_material(NXpiezoelectric_material):
+          exists: optional
+          doc: |
+            The material description and properties of the piezoelectric scanner materials.
+      TEMPERATURE(NXsensor):
```

Comment:
a bit confusing to have temperature here, as we have already discussed some of them in NXenvironment above

Reply:
The temperature in environment group is for single value inside the scan chamber. There is also possibility to collect the temperature at each scan time which is coming from thermo instrument.